linkerd/app/outbound/src/discover.rs

```diff
@@ -21,11 +23,18 @@ pub struct Discovery<T> {
     profile: Option<profiles::Receiver>,
 }
 
+/// Wraps a discovery service with an allowlist of addresses.
+#[derive(Clone, Debug)]
+pub struct WithAllowlist<S> {
+    allow_discovery: AddrMatch,
+    inner: S,
+}
```

Comment:
This strikes me as something that shouldn't be in this module. It's not referenced anywhere in here. Does this even carry weight as a general middlewear?

Reply:
Err, it's not exactly exactly a push switch i guess. It just seems in the uncanny valley between reusable/generic and special-purpose.

Reply:
ah, the reason it's not currently in `linkerd2-proxy-service-profiles` is that the `AddrMatch` stuff is defined in `app-core`, which depends on `service-profiles`, so we would need to move the `AddrMatch` types as well. which i'm willing to do.